R/vcf2bedpe.R

```diff
@@ -0,0 +1,163 @@
+# vcf2bedpe
+
+vcf2bedpe <- function(x, filename = NULL, header = FALSE, other = NULL, verbose = TRUE) {
+  catv('CONVERT VCF TO BEDPE\n')
+  if (!is.null(attr(x, 'vcf')) && attr(x, 'vcf') && all(names(x) == c('header','vcf'))) {
+    x <- x$vcf;
+  }
+  else {
+    catv(' * This is not an vcf!\n')
+    stop();
+  }
+  # use UCSC chromosome convention and subtract one position for start
+  x$CHROM <- gsub('chr', '', x$CHROM);
+  x$POS <- x$POS - 1;
+  if (!'SVTYPE' %in% names(x)) {
+    stop('SVTYPE column must exist in VCF file');
+  }
+  # define columns for bedpe
+  bedpe.cols <- data.frame(CHROM_A = character(0),
+                           START_A = numeric(0),
+                           END_A = numeric(0),
+                           CHROM_B = character(0),
+                           START_B = numeric(0),
+                           END_B = numeric(0),
+                           ID = character(0),
+                           QUAL = character(0),
+                           STRAND_A = character(0),
+                           STRAND_B = character(0),
+                           SVTYPE = character(0)
+                           );
+  # Convert simple breakends with no MATEID
+  simple.bp <- subset(x, SVTYPE != 'BND' & is.na(x$MATEID));
+  if (nrow(simple.bp) > 0) {
+    catv('PROCESSING SIMPLE BREAKENDS\n')
+    coordsA <- adjust.coordinates(simple.bp, 'CIPOS', simple.bp$POS, simple.bp$POS);
+    coordsB <- adjust.coordinates(simple.bp, 'CIEND', simple.bp$END, simple.bp$END);
+    name <- get.name(simple.bp);
+    if (!'CHR2' %in% names(simple.bp)) {
+      simple.bp$CHR2 <- simple.bp$CHROM;
+    }
```

Comment:
What does this do? CHR2 specific? I guess the mate chr is called CHR2?

Reply:
yes CHR2 is the mate chr. In Delly 0.7.8 they use CHR2, even when reporting simple bp (intrachromosomal). This is just to take into account cases without CHR2.

Reply:
Cool. Feel free to add more comments to the script if you like.